src/Microsoft.AspNetCore.SignalR.Client.Core/Internal/SignalRClientLoggerExtensions.cs

```diff
@@ -22,11 +23,11 @@ internal static class SignalRClientLoggerExtensions
         private static readonly Action<ILogger, string, string, string, string, Exception> _issueInvocation =
             LoggerMessage.Define<string, string, string, string>(LogLevel.Trace, new EventId(4, nameof(IssueInvocation)), "Issuing Invocation '{invocationId}': {returnType} {methodName}({args}).");
 
-        private static readonly Action<ILogger, string, Exception> _sendInvocation =
-            LoggerMessage.Define<string>(LogLevel.Debug, new EventId(5, nameof(SendInvocation)), "Sending Invocation '{invocationId}'.");
+        private static readonly Action<ILogger, string, HubMessage, Exception> _sendInvocation =
+            LoggerMessage.Define<string, HubMessage>(LogLevel.Debug, new EventId(5, nameof(SendInvocation)), "Sending Invocation '{invocationId}'. {payload}");
 
-        private static readonly Action<ILogger, string, Exception> _sendInvocationCompleted =
-            LoggerMessage.Define<string>(LogLevel.Debug, new EventId(6, nameof(SendInvocationCompleted)), "Sending Invocation '{invocationId}' completed.");
+        private static readonly Action<ILogger, string, HubMessage, Exception> _sendInvocationCompleted =
+            LoggerMessage.Define<string, HubMessage>(LogLevel.Debug, new EventId(6, nameof(SendInvocationCompleted)), "Sending Invocation '{invocationId}' completed. {payload}");
```

Comment:
Still super unsure about putting message payloads in logs. They potentially have sensitive information in them.

Reply:
Yea the current logging isn’t great but that’s a good point

Reply:
We do it in the JS client too. I'm filing an issue for that because I think this it's generally bad. Let's not add it here though. We can add in attribute **about** the message (type, probably even method name) but not the argument values.

Reply:
https://github.com/aspnet/SignalR/issues/1449